### ISSUE TITLE
Propagate SIMD flags into sub-makes

### ISSUE THREAD

**Issue by solardiz** [openwall/john#5598]
As noticed in #5593, the `CFLAGS` we use to build the `.a` libraries in subdirectories do not include the main build's SIMD flags. This is non-optimal at least in that it doesn't enable usage of the same CPU features, and potentially worse - it may cause slow transitions between VEX-encoded and legacy SSE code.

**Comment by magnumripper** [comment]
The reason for @CFLAGS@ not being "it" is probably because of historical confusion about john.o special treatment, for CPU fallback.

Current **top** Makefile.in
```
CPPFLAGS = @CPPFLAGS@
CFLAGSX = -c @CFLAGS@ @JOHN_NO_SIMD@ @CFLAGS_EXTRA@ @OPENSSL_CFLAGS@ @OPENMP_CFLAGS@ @HAVE_MPI@ @PTHREAD_CFLAGS@ $(CPPFLAGS)
# CFLAGS for use on the main john.c file only
CFLAGS_MAIN = -DAC_BUILT @CC_MAIN_CPU@ $(CFLAGSX)
CFLAGS = -DAC_BUILT @CC_CPU@ @CC_MAIN_CPU@ $(CFLAGSX)
```
We could rework that and edit `configure.ac` so `@CFLAGS@` is all needed, but the easiest fix for the subdirs' Makefile.in's is probably to standardize on something like:
```
CFLAGS = -c -DAC_BUILT @CC_CPU@ @CC_MAIN_CPU@ @CFLAGS@ @JOHN_NO_SIMD@ @CFLAGS_EXTRA@ @OPENSSL_CFLAGS@ @OPENMP_CFLAGS@ @HAVE_MPI@ @PTHREAD_CFLAGS@ @CPPFLAGS@
```
For #5593 I think we should amend `configure.ac` so the `-maes -mpclmul` is added to `@CC_CPU@`, so no change (other than the above) needed in `mbedtls/Makefile.in` for that.

**Comment by magnumripper** [comment]
@solardiz I intend to fix this issue and #5593 while completely ignoring any performance regression [from using intrinsics instead of asm] on some CPU or the other (which could certainly be cared for, but later/separately). OK?

**Comment by solardiz** [comment]
> fix this issue and #5593 while completely ignoring any performance regression [from using intrinsics instead of asm] on some CPU or the other (which could certainly be cared for, but later/separately). OK?

Yes, that's OK with me. Thank you!
